src/contract-variable-monitor/agent.js

```diff
@@ -0,0 +1,157 @@
+  contractAddress,
+  type,
+  severity,
+  protocolName,
+  protocolAbbreviation,
+  developerAbbreviation,
+  thresholdPosition,
+  thresholdPercentLimit,
+  actualPercentChange,
+) {
+  return Finding.fromObject({
+    name: `${protocolName} Contract Variable`,
+    description: `The ${variableName} variable value in the ${contractName} contract had a change`
+      + ` in value over the ${thresholdPosition} threshold limit of ${thresholdPercentLimit}`
+      + ' percent',
+    alertId: `${developerAbbreviation}-${protocolAbbreviation}-CONTRACT-VARIABLE`,
+    type: FindingType[type],
+    severity: FindingSeverity[severity],
+    protocol: protocolName,
+    metadata: {
+      contractName,
+      contractAddress,
+      variableName,
+      thresholdPosition,
+      thresholdPercentLimit: thresholdPercentLimit.toString(),
+      actualPercentChange,
+    },
+  });
+}
+
+const initialize = async (config) => {
+	let agentState = {...config};
+
+	agentState.protocolName = config.protocolName;
+	agentState.protocolAbbreviation = config.protocolAbbreviation;
+	agentState.developerAbbreviation = config.developerAbbreviation;
+	agentState.variableInfoList = [];
+
+	const configEntries = config.contracts;
+	const provider = getEthersProvider();
+
+	// load the contract addresses, abis, and generate an ethers contract for each contract name
+	// listed in the config
+	const contractList = Object.entries(configEntries).map(([name, entry]) => {
+		if (entry.address === undefined) {
+			throw new Error(`No address found in configuration file for '${name}'`);
+		}
+
+		if (entry.abiFile === undefined) {
+			throw new Error(`No ABI file found in configuration file for '${name}'`);
+		}
+
+		const abi = utils.getAbi(config.name, entry.abiFile);
+		const contract = new ethers.Contract(entry.address, abi, provider);
+		return { name, contract, };
+	});
+
+	contractList.forEach((contractEntry) => {
+		const entry = configEntries[contractEntry.name];
+		const { info } = utils.getVariableInfo(entry, contractEntry);
+		agentState.variableInfoList.push(...info);
+	});
+
+	return agentState;
+};
+
+const handleBlock = async (agentState, blockEvent) => {
+	// for each item present in variableInfoList, attempt to invoke the getter method
+	// corresponding to the item's name and make sure it is within the specified threshold percent
+	const variablePromises = agentState.variableInfoList.map(async (variableInfo) => {
+		const variableFindings = [];
+		const {
+			name: variableName,
+			type,
+			severity,
+			contractInfo,
+			upperThresholdPercent,
+			lowerThresholdPercent,
+			minNumElements,
+			pastValues,
+		} = variableInfo;
+		const { name: contractName, contract } = contractInfo;
+
+		let newValue = await contract[variableName]();
+		newValue = new BigNumber(newValue.toString());
+		const averageBN = pastValues.getAverage();
+
+		// check the current number of elements in the pastValues array
+		if (pastValues.getNumElements() >= minNumElements) {
```

Comment:
We use 2 spaces for indent, never tabs:

https://arbitraryexecution.atlassian.net/wiki/spaces/GENERAL/pages/175964352/Coding+Guidelines

Please update all of the code in this PR to fix the extreme indentation and eliminate tabs.